website/src/edit-word-panel.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import { HiPencilAlt } from "react-icons/hi"
+import { IoCheckmarkSharp } from "react-icons/io5"
+import {
+  unstable_Form as Form,
+  unstable_FormInput as FormInput,
+  unstable_FormLabel as FormLabel,
+} from "reakit/Form"
+import { unstable_FormSubmitButton as FormSubmitButton } from "reakit/Form"
+import * as Dailp from "src/graphql/dailp"
+import { IconButton } from "./components"
+import { IconTextButton } from "./components/button"
+import * as css from "./edit-word-panel.css"
+import { useForm } from "./form-context"
+import { usePreferences } from "./preferences-context"
+
+/** Button that allows user to enter edit mode in the word panel, and edit fields of a word. */
+export const EditButton = () => {
+  const { form, isEditing, setIsEditing } = useForm()
+
+  const { cherokeeRepresentation } = usePreferences()
+
+  return (
+    <Form {...form} className={css.form}>
+      {isEditing ? (
+        // Displays a "Cancel" button and "Save" button in editing mode.
+        <>
+          <IconButton
+            className={css.cancelButton}
+            round={false}
+            onClick={() => {
+              setIsEditing(false)
+            }}
+          >
+            Cancel
+          </IconButton>
+
+          <IconTextButton
+            {...form}
+            icon={<IoCheckmarkSharp />}
+            className={css.editPanelButton}
+            as={FormSubmitButton}
+          >
+            Save
+          </IconTextButton>
+        </>
+      ) : (
+        <IconTextButton
+          icon={<HiPencilAlt />}
+          className={css.editPanelButton}
+          onClick={() => {
+            if (cherokeeRepresentation === Dailp.CherokeeOrthography.Taoc) {
+              setIsEditing(true)
+            } else {
+              alert(
+                "Currently, only the linguistic analysis using terms from Tone and Accent in Oklahoma Cherokee (TAOC) is supported for editing. Please update your Cherokee description style in the display settings."
```

Comment:
I want to point out one edge case. If the user is in TAOC mode, they can open the editing panel. Once open, they are able to change preferences out of TAOC mode, but the edit panel remains open. You should see if you can make the edit panel close when settings are changed, disable saving, or some kind of alert if you try to save.